fiona/_meta.pyx

```diff
@@ -0,0 +1,43 @@
+
+include "gdal.pxi"
+
+from fiona._err cimport  exc_wrap_pointer
+from fiona.compat import strencode
+from fiona._shim cimport gdal_open_vector
+from fiona.env import require_gdal_version
+import logging
+
+# This is required otherwise GDALGetDriverByName returns NULL for GDAL 1.x
+from fiona._shim cimport *
+
+
+log = logging.getLogger(__name__)
+
+
+@require_gdal_version('2.0')
+def _get_metadata_item(driver, metadata_item):
+    """Query metadata items
+
+    Parameters
+    ----------
+    driver : str
+        Driver to query
+    metadata_item : str
+        Metadata item to query
+
+    Returns
+    -------
+    str
+        XML of metadata item or empty string
+
+    """
+    cdef char* metadata_c = NULL
+    cdef void *cogr_driver
+
+    metadata = ""
+    cogr_driver = exc_wrap_pointer(GDALGetDriverByName(driver.encode('utf-8')))
+    metadata_c = GDALGetMetadataItem(cogr_driver, strencode(metadata_item), NULL)
```

Comment:
Currently, only the default domain is used (NULL). The domains can be queried with GDALGetMetadataDomainList, however for the driver I tested it with only the default domain (NULL) was reported. I have to admit I don't yet fully understand the domain concept and where it is used.